### PR TITLE
Use Terraform Plugin SDK to Reconcile MRs

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

This PR employs the `controller.noForkExternal` & `controller.noForkAsyncExternal` external clients from upjet to reconcile MRs without forking any Terraform CLI or Terraform provider processes.

Some more details on these clients can be found [here](https://github.com/crossplane/upjet/pull/294).

I have:

- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

**Successful runes of the resources tested with Uptest have been added to the comments below.**

**Manual tests:**
- `Member.administrativeunits`
```
status:
  atProvider:
    administrativeUnitObjectId: 0ae302dc-49fe-445e-b2d1-e0e5448d2e7c
    id: 0ae302dc-49fe-445e-b2d1-e0e5448d2e7c/member/3f061130-d036-4461-8c3e-76a08aabfd08
    memberObjectId: 3f061130-d036-4461-8c3e-76a08aabfd08
  conditions:
  - lastTransitionTime: "2023-11-28T11:17:33Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "2023-11-28T11:17:33Z"
    reason: Success
    status: "True"
    type: LastAsyncOperation
  - lastTransitionTime: "2023-11-28T11:17:33Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
  - lastTransitionTime: "2023-11-28T11:20:57Z"
    reason: UpToDate
    status: "True"
    type: Test
```

- `Invitation.invitations`

Administrator-level permissions are required to test this resource. As far as I understand, I cannot test this resource due to changes made in the organization. But I get the same error in the Azure console as I get in the terminal.

### CODE REVIEW

**Comment by ulucinar** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/administrativeunits/unit.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/app/roleassignment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/applications/certificate.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/applications/preauthorized.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/conditionalaccess/location.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/conditionalaccess/accesspolicy.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/directoryroles/customdirectoryrole.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/directoryroles/role.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/directoryroles/roleassignment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/groups/member.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/policies/claimsmappingpolicy.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipaldelegated/permissiongrant.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/password.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/principal.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/certificate.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/claimsmappingpolicyassignment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/tokensigningcertificate.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/synchronization/job.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/synchronization/secret.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/users/user.yaml"

File: go.mod [hunk 1]
```diff
@@ -3,9 +3,13 @@ module github.com/upbound/provider-azuread
 go 1.19
```
Comment (turkenf) [inline go.mod:3]:
```suggestion
go 1.21
```

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

**Review by turkenf — APPROVED** [review]
Thank you @ulucinar, LGTM.

**Pull request merged**